app/models/katello/host/content_facet.rb

```diff
@@ -29,10 +29,26 @@ class ContentFacet < Katello::Model
       validates_with Validators::ContentViewEnvironmentValidator
 
       def update_repositories_by_paths(paths)
```

Comment:
How about this implementation? I liked the join you added to restrict the content type "just in case", but I simplified some aspects:

```
      def update_repositories_by_paths(paths)                                                                                                                                
        paths = paths.map { |path| path.gsub(/\/pulp\/(repos|deb)\//, '') }                                                                                                  
                                                                                                                                                                             
        repos = Repository.joins(:root).where(RootRepository.table_name => {content_type: [Repository::DEB_TYPE, Repository::YUM_TYPE]},                                     
                                             relative_path: paths)                                                                                                           
                                                                                                                                                                             
        missing_paths = paths - repos.pluck(:relative_path)                                                                                                                  
        missing_paths.each do |repo_path|                                                                                                                                    
          Rails.logger.warn("System #{self.host.name} (#{self.host.id}) requested binding to unknown repo #{repo_path}")                                                     
        end                                                                                                                                                                  
                                                                                                                                                                             
        unless self.bound_repositories.sort == repos.sort                                                                                                                    
          self.bound_repositories = repos                                                                                                                                    
          self.save!                                                                                                                                                         
          self.propagate_yum_repos                                                                                                                                           
          ForemanTasks.async_task(Actions::Katello::Host::GenerateApplicability, [self.host])                                                                                
        end                                                                                                                                                                  
        self.bound_repositories.pluck(:relative_path)                                                                                                                        
      end
```